src/main/java/org/opentestsystem/ap/ims/service/AttachmentService.java

```diff
@@ -16,43 +16,196 @@
+import org.opentestsystem.ap.common.model.ItemTransaction;
+import org.opentestsystem.ap.common.model.ModelConstants;
+import org.opentestsystem.ap.common.model.attachment.AttachmentSelector;
+import org.opentestsystem.ap.common.model.content.GiItemContent;
+import org.opentestsystem.ap.common.security.SecurityUtil;
+import org.opentestsystem.ap.common.security.model.ItemBankUser;
+import org.opentestsystem.ap.common.util.ZipUtil;
+import org.opentestsystem.ap.ims.model.UploadFile;
+
+import static java.lang.String.format;
+import static org.apache.commons.io.FileUtils.deleteQuietly;
+import static org.apache.commons.io.FilenameUtils.removeExtension;
 
 /**
  * Handles file requests like uploading, downloading, and retrieving item files.
  */
 @Slf4j
 @Service
 public class AttachmentService {
-
+    private static final Pattern TRANSLATED_RESOURCE_FILENAME = Pattern.compile("^(.+)_(.+?)$");
     private static final String BULK_PREVIEW_FILE_DELIMITER = ",";
 
     private final ItemManager itemManager;
-
     private final BulkUpdateService bulkUpdateService;
-
     private final SecurityUtil securityUtil;
+    private final DataStoreAttachmentManager attachmentManager;
+    private final AttachmentValidator attachmentValidator;
 
-    public AttachmentService(ItemManager itemManager,
-                             BulkUpdateService bulkUpdateService,
-                             SecurityUtil securityUtil) {
+    public AttachmentService(final ItemManager itemManager,
+                             final BulkUpdateService bulkUpdateService,
+                             final SecurityUtil securityUtil,
+                             final DataStoreAttachmentManager attachmentManager,
+                             final AttachmentValidator attachmentValidator) {
         this.itemManager = itemManager;
         this.bulkUpdateService = bulkUpdateService;
         this.securityUtil = securityUtil;
+        this.attachmentManager = attachmentManager;
+        this.attachmentValidator = attachmentValidator;
     }
 
     // ------------------------------------------------------------------------
 
+    /**
+     * Validates, processes, and uploads the given GI ZIP archive
+     * containing a .GAX grid item definition XML and required
+     * resources (images.)
+     *
+     * @param archiveUpload The uploaded GI archive
+     * @return  The updated GiItemContent referencing the uploaded GI archive resource(s)
+     */
+    public GiItemContent saveGiArchive(final UploadFile archiveUpload) {
+        archiveUpload.validate();
+
+        final ItemBankUser user = securityUtil.getItemBankUser();
+        final GiItem item = (GiItem) itemManager.findItem(user, archiveUpload.getItemId()).getItem();
+        final GiItemContent contentModel = item.getCore().getEn();
+
+        final ItemTransaction userTransaction = itemManager.findUserTransaction(user, archiveUpload.getItemId());
+        if (Objects.isNull(userTransaction)) {
```

Comment:
what does this util do beyond `== null` ?

Reply:
Nothing, but in this project it seems to be the preferred way to null-check.  When in Rome...